Stm32f412 rgt6

honboyxy opened the pull request:
this fork for stm32f412rg

honboyxy committed:
Create STM32F412RG

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Create mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Create mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Create stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Create build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Add files via upload

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Add files via upload

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/build-stm32f412.yaml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Add files via upload

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update board.json

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/stm32/boards/STM32F412RG

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update board.json

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Create stm32f412rg

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Rename stm32f412rg to stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Add files via upload

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Rename build-stm32f412.yaml to build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Create readme.txt

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Add files via upload

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/stm32/boards/STM32F412 directory

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/ports_esp8266.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/ports_rp2.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/ports_esp32.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/ports_cc3200.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/ports_nrf.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/ports_mimxrt.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/ports_samd.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete .github/workflows/ports_zephyr.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/stm32/boards/STM32F412RG/stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/stm32/boards/STM32F412RG/stm32f4xx_hal_conf .h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Add files via upload

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f4xx_hal_conf.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f4xx_hal_conf.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Add files via upload

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f4xx_hal_conf.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f4xx_hal_conf.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update adc.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.mk

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f4xx_hal_conf.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f4xx_hal_conf.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update timer.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Makefile

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Makefile

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/stm32/boards/STM32F412RG/board.json

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/esp8266 directory

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/cc3200 directory

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/mimxrt directory

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/pic16bit directory

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Delete ports/samd directory

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Makefile

Signed-off-by: honboyxy <honboylc@gmail.com>

robert-hh commented:
@honboyxy It seems that you PR got a total mess. So better take the files from the boards/STM32F412RG directory that you have created, close this PR and open a new PR. For adding a new board you changes shall be limited to that directory. Especially you must not touch .gitgub/workflow.
And please do not open a new PR until you have a working firmware for your board, built locally on your PC.

honboyxy committed:
Update build-stm32f412.yml

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update pins.csv

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update pins.csv

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update pins.csv

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update stm32f412rg.ld

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Create MyModule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Rename MyModule.c to mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Rename mymodule.c to Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigboard.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Makefile

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Create Mymodule.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy commented:
> @honboyxy It seems that your PR got a total mess. So better take the files from the boards/STM32F412RG directory that you have created, close this PR and open a new PR. For adding a new board you changes shall be limited to that directory. Especially you must not touch .gitgub/workflow. And please do not open a new PR until you have a working firmware for your board, built locally on your PC.
Hello, I am in the learning stage now and I am very interested in creating micropython. After dozens of times of tossing, I can finally compile it. I don’t know how to build a local compilation environment, so I use the system CI/CD compilation. I want to know, in this case, how to compile the firmware of my own target board without creating a workflow, thank you, in addition, I also want to use the current fork for practice, when I am familiar with it, I will delete this fork and create a new one. I deleted the files because I couldn’t compile it anyway before, so I deleted some files that I thought were useless to me. In addition, I want to understand a problem. STM32F412RG has 1M flash but only 47K can be seen in the computer. I tried to add #define MICROPY_HW_SPIFLASH_SIZE_BITS (32 * 1024 * 1024) // W25Q32 - 32 Mbit (4 MByte), or modify the space of flash_fs in stm32f412rg.ld, but it didn't work. I found that even after I updated the program, the files in the USB disk still existed.

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.c

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update Mymodule.h

Signed-off-by: honboyxy <honboylc@gmail.com>

honboyxy committed:
Update mpconfigport.h

Signed-off-by: honboyxy <honboylc@gmail.com>
